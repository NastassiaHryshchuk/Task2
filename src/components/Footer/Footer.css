.footer {
    background-color: rgb(53, 53, 53);
    padding: 5px 0px;
    flex-shrink: 0;
}