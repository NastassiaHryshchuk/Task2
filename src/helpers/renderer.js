import React from 'react';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Routes from '../Routes/Routes';

export default(req, store) => {
  const content = renderToString(
    <Provider store={store}>
      <StaticRouter location={req.path} context={{}}>
        <Routes />
      </StaticRouter>
    </Provider>,
  );

  return `
    <!DOCTYPE html>
    <html>
      <head>
        <title>SSR</title>
      </head>
      <body>
        <div id="app">${content}</div>
        <script>
          window.INITIAL_STATE = ${JSON.stringify(store.getState())}
        </script>
        <script src="bundle.js"></script>
      </body>
    </html>
  `;
};
