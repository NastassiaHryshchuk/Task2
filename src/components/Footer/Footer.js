import React, { Component } from 'react';
import './Footer.css';

class Footer extends Component {
  render() {
    return (
        <footer className="footer">
            <div className="wrap">
                <p className="logo">netflixroulette</p>
            </div>
         </footer>
    );
  }
}

export default Footer;
