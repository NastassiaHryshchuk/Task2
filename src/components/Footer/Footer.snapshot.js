import React from 'react';
import Footer from './Footer';
import renderer from 'react-test-renderer';

describe('Footer Snapshot', () => {
  test('renders', () => {
    const component = renderer.create(
      <Footer />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

