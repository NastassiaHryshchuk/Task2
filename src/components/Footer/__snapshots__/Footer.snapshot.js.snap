// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot renders 1`] = `
<footer
  className="footer"
>
  <div
    className="wrap"
  >
    <a
      className="logo"
      href="#"
    >
      netflixroulette
    </a>
  </div>
</footer>
`;
